### PR TITLE
Fix node freeze after re-enabling consensus [ERC-3111]

### PR DESCRIPTION
**Pull request by alekseysidorov**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -464,6 +464,8 @@ Key points:
 
 - #1568: Make DispatcherSchema merkelized again [ECR-3810]
 
+- #1592: Fix node free after re-enable consensus [ERC-3111]
```
Comment (popzxc) [inline CHANGELOG.md]: "Fix node free" doesn't sound quite correct in English.

File: exonum/src/node/consensus.rs [hunk 2]
```diff
@@ -978,6 +986,11 @@ impl NodeHandler {
     /// Requests a block for the next height from all peers with a bigger height. Called when the
     /// node tries to catch up with other nodes' height.
     pub(crate) fn request_next_block(&mut self) {
+        if !self.is_enabled {
+            info!("Do not send a request for the next block because the node is paused.",);
```
Comment (vitvakatu) [inline exonum/src/node/consensus.rs]: Are you sure the paused node should log such messages with `info` level?

File: CHANGELOG.md (hunk 1 above)
Comment (alekseysidorov) [reply CHANGELOG.md]: Mac autocorrect thinks different.

File: CHANGELOG.md [hunk 3]
```diff
@@ -464,6 +464,8 @@ Key points:
 
 - #1568: Make DispatcherSchema merkelized again [ECR-3810]
 
+- #1592: Fix node freeze after re-enable consensus [ERC-3111]
```
Comment (vitvakatu) [inline CHANGELOG.md]:
```suggestion
- #1592: Fix node freeze after re-enabling consensus [ERC-3111]
```

File: exonum/src/node/consensus.rs [hunk 4]
```diff
@@ -857,6 +857,14 @@ impl NodeHandler {
         if let Some(peer) = self.state.retry(data, peer) {
             self.add_request_timeout(data.clone(), Some(peer));
 
+            if !self.is_enabled {
+                trace!(
+                    "Do not send a request {:?} because the node is paused.",
```
Comment (slowli) [inline exonum/src/node/consensus.rs]: Nit: "Not sending a request..." makes more sense from the grammar perspective, I believe.

File: exonum/src/node/consensus.rs [hunk 5]
```diff
@@ -978,6 +986,11 @@ impl NodeHandler {
     /// Requests a block for the next height from all peers with a bigger height. Called when the
     /// node tries to catch up with other nodes' height.
     pub(crate) fn request_next_block(&mut self) {
+        if !self.is_enabled {
+            trace!("Do not send a request for the next block because the node is paused.",);
```
Comment (slowli) [inline exonum/src/node/consensus.rs]: Nit: Bogus comma; also "Not sending a request..." would make more sense grammatically.

**Pull request merged**